cpp/include/cudf/detail/gather.cuh

```diff
@@ -245,7 +245,7 @@ struct column_gatherer_impl<dictionary32, MapItType> {
     auto indices_column = std::make_unique<column>(data_type{INT32},
                                                    static_cast<size_type>(output_count),
                                                    std::move(*(contents.data.release())),
-                                                   rmm::device_buffer{},
+                                                   rmm::device_buffer(0, stream, mr),
```

Comment:
This seems unnecessary.

Reply:
I'm sure many of my changes from default `device_buffer` ctor to `(0, stream, mr)` are pointless ATM. Some may become relevant if the output changes such that resize is called on the empty buffer (not really possible here). 
Regardless, IMO it does not hurt to call the right ctor here even if it does not affect the allocation. Also, I find it easier to use mr for all returned objects than to consider if an allocation really happens. 
Do you see a downside to this change?

Reply:
Yeah, that about matches what I expected. 

> Do you see a downside to this change?

Not really, just more verbose.